vignettes/ABN_inference.Rmd

```diff
@@ -15,201 +17,324 @@ knitr::opts_chunk$set(
+
+    - the variance $var(X|B)$ is:
+$$var(X|B) = \mathbb{E}(X^2|B) - \left(\mathbb{E}(X|B)^2\right),$$
+with: 
+$$\mathbb{E}(X^2|B) =  \frac{\int X^2 \mathbb{P}(B|X) \mathbb{P}(X) dX}{\int \mathbb{P}(B|X) \mathbb{P}(X) dX}.$$ 
+
+
+In the formula above, we use the estimated distribution of $X$ from Step 1 as a prior:
+
+- if $X$ is Binomial, we have $X \sim B(p_X)$, where $p_X$ is estimated in Step 1. 
+
+- if $X$ is Gaussian, we have $X \sim N(\mu_X,\sigma_X^2)$, where $\mu_X$ and $\sigma_X^2$ are estimated in Step 1. 
+
+- if $X$ is Poisson, we have $X \sim P(\lambda_X)$, where $\lambda_X$ is estimated in Step 1. 
+
+The conditional probability $\mathbb{P}(B|X)$ is derived from the likelihood function $L(B|X)$: 
+
+- if $B$ is Binomial, $B \sim B(p_B)$, with $p_B = \frac{1}{1+\exp \left( - (\beta_{B0} + \beta_{B1} X + \beta_{B2} C)\right)},$ 
+
+- if $B$ is Gaussian, $B \sim N(\mu_B,\sigma_B^2)$, with $\mu_B = \beta_{B0} + \beta_{B1} X+ \beta_{B2} C$, 
+ <span style="color: red;">What to do with the variance? I use the estimated variance for $B$.</span>
+
+- if $B$ is Poisson, $B \sim P(\lambda_B)$, with $\lambda_B = \exp \left( \beta_{B0} + \beta_{B1} X + \beta_{B2} C \right)$.
+
+Here, for simplicity reasons, $C$ represents the unique parent of $B$, aside from $X$.
+
+Substitute the known values of the parents $\mathscr{C}$ of the child $B$ into the expression to estimate the conditional distribution.
+
+    
+<span style="color: red;">We are not really using the fact that $B$ is an evidence as we consider the link between $B$ and $C$ (?) </span>
+
+
+If $X$ has more than one child, repeat the same procedure for all children $B$ of $X$. In this case, for each type of distribution for $X$, the estimates are averaged over the children of $X$:
+
+- if $X$ is Poisson:
+$\mathbb{E}(X|\mathscr{A},\mathscr{B},\mathscr{C}) = \frac{1}{\# \mathscr{B}} \sum_{B_j \in\mathscr{B}}  \mathbb{E}(X|\mathscr{A},B_j,\mathscr{C}_j),$
+
+- if $X$ is Gaussian:
+
+    - the expected value $\mathbb{E}(X|\mathscr{A},\mathscr{B},\mathscr{C})$ is given by:
+$\mathbb{E}(X|\mathscr{A},\mathscr{B},\mathscr{C}) = \frac{1}{\# \mathscr{B}} \sum_{B_j \in\mathscr{B}}  \mathbb{E}(X|\mathscr{A},B_j,\mathscr{C}_j),$
+
+    - the variance  $var(X|\mathscr{A},\mathscr{B},\mathscr{C})$ is given by:
+    $var(X|\mathscr{A},\mathscr{B},\mathscr{C}) = \frac{1}{\# \mathscr{B}^2} \sum_{B_j \in\mathscr{B}}  var(X|\mathscr{A},B_j,\mathscr{C}_j),$
+    
+- if $X$ is Binomial:
+
+    - $\mathbb{P}(X=1|\mathscr{A},\mathscr{B},\mathscr{C}) = \frac{1}{\# \mathscr{B}} \sum_{B_j \in\mathscr{B}}  \mathbb{P}(X=1|\mathscr{A},B_j,\mathscr{C}_j),$
+    
+    - $\mathbb{P}(X=0|\mathscr{A},\mathscr{B},\mathscr{C}) = 1-\mathbb{P}(X=1|\mathscr{A},\mathscr{B},\mathscr{C}),$
+
+where $\mathscr{C}_j$ denotes the set of parents of $B_j$. 
```

Comment:
Is it correct that it is assumed here that all children $B_j$ are conditionally independent of each other, i.e., $\mathscr{C} \notin \mathscr{B}$?